sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1288,4 +1299,71 @@ export class ApplicationService extends RecordDataModelService<Application> {
     }
     return applicationQuery.getOne();
   }
+
+  /**
+   * Validation for application overlapping dates or Pending PIR.
+   * This validation can be disabled by setting BYPASS_APPLICATION_SUBMIT_VALIDATIONS to true in .env file.
+   * @param applicationId
+   * @param lastName
+   * @param userId
+   * @param sin
+   * @param birthDate
+   * @param studyStartDate
+   * @param studyEndDate
+   * @param bypassApplicationSubmitValidations
+   */
+  async validateOverlappingDatesAndPIR(
+    applicationId: number,
+    lastName: string,
+    userId: number,
+    sin: string,
+    birthDate: Date,
+    studyStartDate: Date,
+    studyEndDate: Date,
+  ): Promise<void> {
+    if (!this.bypassApplicationSubmitValidations) {
+      const existingOverlapApplication = this.validatePIRAndDateOverlap(
+        userId,
+        new Date(studyStartDate),
```

Comment:
do we need `new Date` transformation? the type is already date right? line 1321 or 1222